spec/image_spec.rb

```diff
@@ -95,72 +81,117 @@
       before do
-        generate_test_image(TEST_TIF_INPUT_FILE, color: 'Bilevel', image_type: 'Grayscale', profile: '')
+        generate_test_image(TEST_TIF_INPUT_FILE, color: 'grey', bands: 1)
       end
 
       it 'creates grayscale jp2' do
         expect(File).to exist TEST_TIF_INPUT_FILE
         expect(File).not_to exist TEST_JP2_OUTPUT_FILE
         expect(assembly_image).not_to have_color_profile
+        expect(assembly_image.exif.samplesperpixel).to be 1
+        expect(assembly_image.exif.bitspersample).to be 8
+        expect(assembly_image).to be_a_valid_image
+        expect(assembly_image).to be_jp2able
         result = assembly_image.create_jp2(output: TEST_JP2_OUTPUT_FILE)
         expect(TEST_JP2_OUTPUT_FILE).to be_a_jp2
-        expect(result.exif.colorspace).to eq 'Grayscale'
+        expect(result).to be_a_kind_of described_class
+        expect(result.path).to eq TEST_JP2_OUTPUT_FILE
+        # expect(result.exif.colorspace).to eq 'Grayscale'
       end
     end
 
     context 'when given a color tif but with greyscale image data (1 channel and 8 bits per pixel)' do
       before do
-        generate_test_image(TEST_TIF_INPUT_FILE, color: 'Grayscale', image_type: 'TrueColor', profile: '')
+        generate_test_image(TEST_TIF_INPUT_FILE, color: 'grey')
       end
 
       it 'creates color jp2' do
         expect(File).to exist TEST_TIF_INPUT_FILE
         expect(File).not_to exist TEST_JP2_OUTPUT_FILE
+        expect(assembly_image.exif.samplesperpixel).to be 3
+        expect(assembly_image.exif.bitspersample).to eql '8 8 8'
+        expect(assembly_image).to be_a_valid_image
+        expect(assembly_image).to be_jp2able
         expect(assembly_image).not_to have_color_profile
         result = assembly_image.create_jp2(output: TEST_JP2_OUTPUT_FILE)
+        expect(result).to be_a_kind_of described_class
+        expect(result.path).to eq TEST_JP2_OUTPUT_FILE
         expect(TEST_JP2_OUTPUT_FILE).to be_a_jp2
         expect(result.exif.colorspace).to eq 'sRGB'
       end
     end
 
+    context 'when given a cmyk tif' do
+      before do
+        generate_test_image(TEST_TIF_INPUT_FILE, color: 'cmyk', cg_type: 'cmyk', profile: 'cmyk', bands: 4)
+      end
+
+      it 'creates jp2' do
+        expect(File).to exist TEST_TIF_INPUT_FILE
+        expect(File).not_to exist TEST_JP2_OUTPUT_FILE
+        expect(assembly_image.exif.samplesperpixel).to be 4
+        expect(assembly_image.exif.bitspersample).to eql '8 8 8 8'
+        expect(assembly_image).to be_a_valid_image
+        expect(assembly_image).to be_jp2able
+        expect(assembly_image).to have_color_profile
+        result = assembly_image.create_jp2(output: TEST_JP2_OUTPUT_FILE)
+        expect(result).to be_a_kind_of described_class
+        expect(result.path).to eq TEST_JP2_OUTPUT_FILE
+        expect(TEST_JP2_OUTPUT_FILE).to be_a_jp2
+        # expect(result.exif.colorspace).to eq 'sRGB'
+      end
```

Comment:
same comment as above

Reply:
this is a cmyk image ... is this something we'll handle as part of issue #55 ?

Reply:
yeah, needs investigation by Tony (noted in the PR description above) ... likely an issue with the test image generation code in spec_helper

Reply:
This one gives me pause on merging -- are we losing a capability we formerly had?

Reply:
Adjusted test to use a ruby-vips image instead of exif